### PR TITLE
Deprecation notice for calc and eval

### PR DESCRIPTION
**Pull request by breville**
This shows some deprecation notices for `"algebra_game"`, `"calc"`, and `"eval"` project types.  

If a signed-in user (student or teacher) has created a project of one of these types, they'll see a notice on their `/home` and `/projects` pages.

If someone is viewing a shared project of one of these types, they'll see a notice below the "How It Works" button.

This is very temporary code, and can be removed again once the related content is deprecated.

### /home

Screenshot 2023-07-11 at 5 55 30 PM

### /projects

Screenshot 2023-07-11 at 5 56 02 PM

### Shared project

Screenshot 2023-07-11 at 5 54 53 PM

### CODE REVIEW

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -747,6 +747,9 @@
   "deleteUsedImage": "{name} is used in {value, plural, one {1 place} other {# places}} in this app. If you delete it, the image will not appear in any of those places. Are you sure you want to delete it?",
   "deleting": "Deleting",
   "deletingWithEllipsis": "Deleting...",
+  "deprecatedCalcAndEvalBrief": "Within the next month, this project will be removed.",
```
Comment (fisher-alice) [inline apps/i18n/common/en_us.json]: 'removed permanently'?

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -747,6 +747,9 @@
...
+  "deprecatedCalcAndEvalWarning": "Calc and Eval project types are going away!",
+  "deprecatedCalcAndEvalDetails": "Code.org is the process of fully removing CS in Algebra from our course catalog. Within the next month, all Calc and Eval projects types are going to be removed from the project gallery and your personal projects store. If you want to keep your code, copy it into a text editor before the end of July.",
```
Comment (fisher-alice) [inline apps/i18n/common/en_us.json]: typo: 'Code.org is IN the process...'

Comment (fisher-alice) [reply apps/i18n/common/en_us.json]: 'project types'

File: dashboard/app/controllers/home_controller.rb [hunk 3]
```diff
@@ -112,6 +112,7 @@ def terms_and_privacy
 
     current_user_permissions = UserPermission.where(user_id: current_user.id).pluck(:permission)
     @homepage_data[:showStudentAsVerifiedTeacherWarning] = current_user.student? && current_user_permissions.include?(UserPermission::AUTHORIZED_TEACHER)
+    @homepage_data[:showDeprecatedCalcAndEvalWarning] = current_user.student? && ProjectsList.user_has_project_type(current_user.id, ['algebra_game', 'calc', 'eval'])
```
Comment (fisher-alice) [inline dashboard/app/controllers/home_controller.rb]: Will the deprecation warning not be displayed for teachers with these types of projects?

**Review by fisher-alice — APPROVED** [review]
LGTM! Just a question and a couple nits.

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (samantha-code) [reply apps/i18n/common/en_us.json]: Lets add it

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (samantha-code) [reply apps/i18n/common/en_us.json]: Yup! lets change these both!

File: dashboard/app/controllers/home_controller.rb (hunk 3 above)
Comment (samantha-code) [reply dashboard/app/controllers/home_controller.rb]: What kind of delay would it have to add it for teachers as well?

Comment (fisher-alice) [reply dashboard/app/controllers/home_controller.rb]: I think it's just tweaking the condition above.

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (breville) [reply apps/i18n/common/en_us.json]: Also should we use "curriculum catalog" to match the name of our recently updated catalog page?

Comment (breville) [reply apps/i18n/common/en_us.json]: Also, do students know what a "text editor" is, and is there actually a way to copy this code into one given that it's block-based?

File: dashboard/app/controllers/home_controller.rb (hunk 3 above)
Comment (breville) [reply dashboard/app/controllers/home_controller.rb]: Yes, we can just remove the check for being a student, and then it should show for both students and teachers.

File: apps/src/templates/studioHomepages/StudentHomepage.jsx
Comment (breville) [inline apps/src/templates/studioHomepages/StudentHomepage.jsx:1]: @samantha-code Do we want to make the same change for the teacher homepage?

File: dashboard/app/controllers/home_controller.rb (hunk 3 above)
Comment (breville) [reply dashboard/app/controllers/home_controller.rb]: Oh, this check is used for the user's homepage. If we want the notification to show on the teacher's homepage as well, we'll need to modify the client-side code that renders their homepage.

File: dashboard/app/views/projects/index.html.haml [hunk 4]
```diff
@@ -7,10 +7,11 @@
 
   if current_user
     projects_data[:canViewAdvancedTools] = !(current_user.under_13? && current_user.terms_version.nil?)
-    projects_data[:canShare] = !current_user.sharing_disabled?
+    projects_data[:canShare] = !current_user&.sharing_disabled?
   end
 
   projects_data[:specialAnnouncement] = Announcements.get_announcement_for_page("/projects")
+  projects_data[:showDeprecatedCalcAndEvalWarning] = current_user.student? && ProjectsList.user_has_project_type(current_user.id, ['algebra_game', 'calc', 'eval'])
```
Comment (breville) [inline dashboard/app/views/projects/index.html.haml]: And we want the notification to show on a teacher's `/projects` page as well?

File: dashboard/app/controllers/home_controller.rb (hunk 3 above)
Comment (breville) [reply dashboard/app/controllers/home_controller.rb]: I've updated the change to also show the notification on a teacher's `/home` and /projects` page.

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (fisher-alice) [reply apps/i18n/common/en_us.json]: I was actually wondering the same thing about the 'text editor'. Maybe something like 'If you want to keep your code, you must make a personal copy of it before the end of July'?

Comment (breville) [reply apps/i18n/common/en_us.json]: @samantha-code  ^

Comment (samantha-code) [reply apps/i18n/common/en_us.json]: Code.org is the process of fully removing CS in Algebra from our curriculum catalog. Within the next month, all Calc and Eval projects types are going to be removed from the project gallery and your personal projects store. If you want to keep your code, you must make a personal copy before the end of July

**Review by fisher-alice — APPROVED** [review]
Thanks for the updates, Brendan! And thanks @samantha-code for taking another look.

**Pull request merged**